packages/dd-trace/src/debugger/devtools_client/index.js

```diff
@@ -20,9 +21,33 @@ const threadName = parentThreadId === 0 ? 'MainThread' : `WorkerThread:${parentT
 session.on('Debugger.paused', async ({ params }) => {
   const start = process.hrtime.bigint()
   const timestamp = Date.now()
-  const probes = params.hitBreakpoints.map((id) => breakpoints.get(id))
+
+  let captureSnapshotForProbe = null
+  let maxReferenceDepth, maxLength
+  const probes = params.hitBreakpoints.map((id) => {
+    const probe = breakpoints.get(id)
+    if (probe.captureSnapshot) {
+      captureSnapshotForProbe = probe
+      maxReferenceDepth = highestOrUndefined(probe.capture.maxReferenceDepth, maxReferenceDepth)
+      maxLength = highestOrUndefined(probe.capture.maxLength, maxLength)
+    }
+    return probe
+  })
+
+  let processLocalState
+  if (captureSnapshotForProbe !== null) {
+    try {
+      // TODO: Create unique states for each affected probe based on that probes unique `capture` settings (DEBUG-2863)
+      processLocalState = await getLocalStateForCallFrame(params.callFrames[0], { maxReferenceDepth, maxLength })
+    } catch (err) {
+      // TODO: This error is not sepcifically tied to a specifc probe, but to all probes with `captureSnapshot: true`.
```

Comment:
nit: two "specific"s sound redundant, but if you want to keep them, they have a typo 😄 
```suggestion
      // TODO: This error is not tied specifically to a specific probe, but to all probes with `captureSnapshot: true`.
```

Reply:
I'll fix the comment, thanks 👍